lib/galaxy/job_execution/setup.py

```diff
@@ -122,9 +121,14 @@ def __init__(
         self.tool_source = tool_source
         self.tool_source_class = tool_source_class
         self._output_paths: Optional[OutputPaths] = None
+        self._output_paths_for_job_object_id: Optional[int] = None
         self._output_hdas_and_paths: Optional[OutputHdasAndType] = None
         self._dataset_path_rewriter: Optional[DatasetPathRewriter] = None
 
+    @property
+    def job(self):
+        return self.sa_session.query(Job).get(self.job_id)
```

Comment:
If we want transparent and cached access to the the job but a fresh job per thread - we could use a threading.local to cache this just for the thread it is in https://stackoverflow.com/questions/1408171/thread-local-storage-in-python. I think we use that pattern a few places in the code base.

Reply:
Ah, yes, of course, that's a good idea

Reply:
(I'll keep that here the same, since getting from the session is cheap, but it's a good idea for `_output_paths_for_job_object_id`)